### ISSUE TITLE
Mail deleted in K9Mail but not on the remote server

### ISSUE THREAD

**Issue by alxnbl** [thunderbird/thunderbird-android#6292]
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same bug report.
- [X] I will describe the problem with as much detail as possible.

### App version

6.301

### Where did you get the app from?

FDroid

### Android version

Android 11; /e/ OS 1.3-20220905216075

### Device model

Fairphone 4

### Steps to reproduce

1. Run K9Mail
2. Force Mail sync
3. Delete a 1st mail -> mail is removed from K9Mail and is removed from the Server (check from my mail provider webapp)
4. Delete a 2nd mail -> mail is removed from K9Mail but not from the Server
5. Force Mail sync -> 2nd mail is downloaded from the server and reappear in K9Mail

Other informations : 
- In parallel I run several tests with Thunderbird connected to the same account and never been able to reproduce. 
- I have a second mail account configured in K9Mail hosted by an other provider (OVH), and I never encounter the problem. Seams really something specific to my mail provider (Mailfence) or to my mail account...

### Expected behavior

I expect the mail been removed on the server

### Actual behavior

The mail is not removed on the server

### Logs

[k9-log4.txt](https://github.com/thundernest/k-9/files/9540318/k9-log4.txt)

**Comment by alxnbl** [comment]
Hop it can help, do not hesitate to ask me additional info if needed.
Thank you for you work!

**Comment by cketti** [comment]
The log only shows one message being deleted.

First the message is copied to the Trash folder.
```text
>>> 12 UID COPY 4375 "Trash"
<<<#12# [OK, [COPYUID, 804386077, 4375, 2289], COPY Completed]
```

Then the original message is marked as deleted (in the source folder).
```text
>>> 14 UID STORE 4375 +FLAGS.SILENT (\Deleted)
<<<#14# [OK, STORE Completed]
```

K-9 Mail doesn't seem to send an EXPUNGE command that actually removes the message marked as deleted from the server. Did you change *Settings → [Account] → Fetching mail → Erase deleted messages on server* from *Immediately* to something else?

The above shouldn't matter. K-9 Mail will not download messages that are marked as deleted. However, when getting the list of non-deleted messages, the original message (with the UID `4375`) is included.

```text
>>> 22 UID SEARCH 1:94 NOT DELETED
<<<#null# [SEARCH, …, 4375, …]
<<<#22# [OK, SEARCH completed]
```

This is either a bug in the server software, or some other client removed the `\Deleted` flag on the message. 

Please contact your email provider and ask them to look into this.

**Comment by alxnbl** [comment]
Thank you for the quick answer.

You're right regarding my "Erase deleted messages on server" config that was set to "During email fetching" (or a similar label).
I set the setting to "Immediately", close and start the app, then delete the mail and I observed the same behavior. 
Please note that I found an exception in the log, but it seams related to my OVH account, not the Mailfence account concerned by the bug.
Log file : [k9-log6.txt](https://github.com/thundernest/k-9/files/9540420/k9-log6.txt)

If you confirm your first analysis I will contact Mailfence. But do you have any clue why the issue do not occurs on Thunderbird ? I mean If Mailfence do it the wrong way, It should do it badly with both K9Mail and Thunderbird isn't it ? Except if Thunderbird and K9Mail do not rely on the same API to execute the deletion... Or maybe my analysis is totally wrong ;).

Again thank you for your time. And by the way very happy you are joining the Thunderbird team !

**Comment by cketti** [comment]
This is a duplicate of #6131

**Comment by alxnbl** [comment]
Thanks, indeed looks very similar. Including issue with "Mark as read". I will contact Mailfence and consider switching to an other provider.
